python/triton/ops/flash_attention.py

```diff
@@ -370,6 +375,11 @@ def forward(ctx, q, k, v, causal, sm_scale, sequence_parallel=False):
             raise RuntimeError("Flash attention currently only supported for compute capability >= 80")
         BLOCK_M = 128
         BLOCK_N = 64
+
+        if is_hip():
+            # Some accuracy issues observed on hip with BLOCK_M = 128.
+            BLOCK_M = 64
```

Comment:
It seems the tests fail when D_HEAD=16 + dtype=float16. Can you dig more into the mismatch issue?